### ISSUE TITLE
解决Ubuntu启动错误——kernel panic not syncing vfs unable to mount root fs on unknown-block 0 0 – error

### ISSUE THREAD

**Issue by zhongdeming428** [zhongdeming428/Blog#18]
![2018-06-07 14-09-16](https://user-images.githubusercontent.com/25274581/41081372-3188bfac-6a5d-11e8-8eb2-3a39c550a12b.png)

最近在倒腾Ubuntu，然后想着怎么美化一下界面，于是照着网上的教程整了一下`Flatabulous`这个软件，然后好像`/boot`就满了。关机之后再开机就出现了如题所述的错误，无法开机，也无法进入`recovery mode`,整个系统都感觉没救了。找了一些答案好多都说需要使用`live CD`的方式启动，太麻烦了。后来找到了一个靠谱点的教程，现在记录一下。

**首先，最好对你的系统做个备份，或者导出，确保意外发生的时候，给你的造成的损失是最小的。**

出现这种错误的原因多半是`/boot`盘空间不够，你的`Ubuntu`内核做了许多的更新，而默认情况下，`Ubuntu`将保留旧的内核并将它们添加到可在`Grub2`启动加载器菜单中启动的可用内核列表中。

你可以通过以下命令来确定你的`/boot`是否已经爆满：

```
df -h
```

现在的情况是系统已经无法进入，连修复模式也没办法开启，所以我们要想办法。在进入引导界面的时候，可以选择`Ubuntu高级选项`，然后选择一个非修复模式的系统进入。当然不是所有情况下都能顺利进入系统，这个时候你只能祈祷能够顺利进入登录界面咯。

登录之后，打开命令行工具，使用以下命令查看你的当前内核：

```
uname -a
```

然后把输出结果记在你的小本本上吧，等会千万不能删除这个内核。

接下来就是通过以下命令查看你的机器上安装的所有内核：

```
dpkg --list | -grep linux-image
```

找到所有比当前内核版本低的内核。当你知道要删除哪些内核时，请继续在下面将其删除。运行以下命令删除您选择的内核：

```
sudo apt-get purge linux-image-xxxx-generic
```

或者：

```
sudo apt-get purge linux-image-extra-xxx-xx-generic
```

最后，运行下面的命令更新grub2：

```
sudo update-grub2
```

最后的最后，重启你的机器：

```
sudo reboot
```

这次在登录的时候，就可以直接使用第一个选项`Ubuntu`进行登录了。

原文链接：[Fix ubuntu when the OS will not boot – kernel panic – kernel panic not syncing vfs unable to mount root fs on unknown-block 0 0 – error /boot full remove old kernels from command line](https://jasoncoltrin.com/2017/01/13/fix-ubuntu-when-the-os-will-not-boot-kernel-panic-kernel-panic-not-syncing-vfs-unable-to-mount-root-fs-on-unknown-block-0-0-error-boot-full-remove-old-kernels-from-command-line/)

**Comment by Zhefei777** [comment]
感谢楼主~

**Comment by zhongdeming428** [comment]
@Zhefei777 能帮到你就行^-^

**Comment by Dawnstar2018** [comment]
it does not  seem to work

**Comment by zhongdeming428** [comment]
@Dawnstar2018 
Did u read this article ?

https://jasoncoltrin.com/2017/01/13/fix-ubuntu-when-the-os-will-not-boot-kernel-panic-kernel-panic-not-syncing-vfs-unable-to-mount-root-fs-on-unknown-block-0-0-error-boot-full-remove-old-kernels-from-command-line/

what's ur problem?

**Comment by Dawnstar2018** [comment]
yes,i read it.but when i do all the steps as above,i still can't enter the first choice ubuntu

**Comment by zhongdeming428** [comment]
@Dawnstar2018 I'm not familiar with Linux, too 😭 .Sorry for nothing help you
